src/cloud-api-adaptor/install/overlays/gcp/kustomization.yaml

```diff
@@ -0,0 +1,55 @@
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+
+resources:
+- ../../yamls
+
+images:
+- name: cloud-api-adaptor
+  newName: 192.168.122.1:5000/cloud-api-adaptor # change image if needed
+  newTag: latest
+
+generatorOptions:
+  disableNameSuffixHash: true
+
+configMapGenerator:
+- name: peer-pods-cm
+  namespace: confidential-containers-system
+  literals:
+  - CLOUD_PROVIDER="gcp"
+  #- PAUSE_IMAGE="" # Uncomment and set if you want to use a specific pause image
+  #- VXLAN_PORT="" # Uncomment and set if you want to use a specific vxlan port. Defaults to 4789
+  - PODVM_IMAGE_NAME="" # set from step "Build Pod VM Image" in gcp/README.md
+  - GCP_PROJECT_ID="" # set
+  - GCP_ZONE="" # set e.g. "us-west1-a"
+  - GCP_MACHINE_TYPE="e2-medium" # replace if needed. caa defaults to e2-medium
+  - GCP_NETWORK="global/networks/default" # replace if needed.
+##TLS_SETTINGS
+  #- CACERT_FILE="/etc/certificates/ca.crt" # for TLS
+  #- CERT_FILE="/etc/certificates/client.crt" # for TLS
+  #- CERT_KEY="/etc/certificates/client.key" # for TLS
+  #- TLS_SKIP_VERIFY="" # for testing only
+##TLS_SETTINGS
+
+secretGenerator:
+- name: auth-json-secret
+  namespace: confidential-containers-system
+  files:
+  #- auth.json # set - path to auth.json pull credentials file
+- name: peer-pods-secret
+  namespace: confidential-containers-system
+  files:
+  - GCP_CREDENTIALS # make sure this file has the application credentials. You can reuse the Packer creds created in "Build Pod VM Image"
```

Comment:
Can env file be used to provide the secrets followed by a local .gitignore file under `overlays/gcp`, similar to aws and azure? This will make it consistent.

https://github.com/confidential-containers/cloud-api-adaptor/blob/main/src/cloud-api-adaptor/install/overlays/aws/kustomization.yaml#L55
https://github.com/confidential-containers/cloud-api-adaptor/blob/main/src/cloud-api-adaptor/install/overlays/azure/kustomization.yaml#L62

Reply:
Yes.. I considered using env variables for GCP credentials. But GCP typically uses the credentials JSON files, which seems to be the standard among tools, docs and users. I think it might be simpler for users to stick with that. Env files make sense for Azure, but not sure if the same is valid for GCP. Converting each key (specially things like `private_key`) from the GCP credentials to individual variables could be a bit cumbersome.

Let’s discuss if you see it differently.

Reply:
Hmm, I see that the gcp code here expects the creds to be a json file. So I think it's ok for now. We can think about this later.